Homework.playground/Contents.swift

```diff
@@ -1,3 +1,88 @@
-import Cocoa
+import Foundation
 
-var str = "Hello, playground"
+func mean(array: [Int]) -> Double? {
+    guard !array.isEmpty else {
+        return nil
+    }
+    let sum = array.reduce(0, +)
+    return Double(sum) / Double(array.count)
+}
+
+func median(array: [Int]) -> Int? {
+    guard !array.isEmpty else {
+        return nil
+    }
+    let sortedArray = quickSort(array: array)
+    let count = sortedArray.count
+    if count.isMultiple(of: 2) {
+        let rightNum = sortedArray[count/2]
+        let leftNum = sortedArray[count/2 - 1]
+        return (leftNum + rightNum)/2
+    } else {
+        return sortedArray[count/2]
+    }
+    
+}
+
+func mode(array: [Int]) -> [Int] {
```

Comment:
According the your implementation the sequence `[1,2,3]` has 3 modes: `1`, `2`, `3`. What does it mean?

Reply:
You wanna say if all numbers in the array are mode, then there are no mode numbers? And the return value of `[1,2,3]` needs to be `nil` (for example)?

Reply:
Yes, I think for the user, it's more simple to know we don't have an explicit mode, instead of returning all the values. These values can be retrieved from the initial array.